src/coreclr/src/utilcode/clrhost.cpp

```diff
@@ -27,12 +27,13 @@ void* GetClrModuleBase()
 #else // HOST_WINDOWS
     // PAL_GetSymbolModuleBase defers to dladdr, which is typically a hash lookup through symbols.
     // It should be fairly fast, however it may take a loader lock, so we will cache the result.
-    if (!pImageBase)
+    void* pRet = VolatileLoadWithoutBarrier(&pImageBase);
```

Comment:
How about just making the pImageBase volatile (I mean, using the C++ volatile keyword) instead? We do that for many cases like this.

Reply:
I thought about that, but I see relatively very few uses of `volatile` outside of code that implements concurrency helpers. I think it is discouraged since it means a different thing in C#.

`VolatileLoadWithoutBarrier` seems a lot more popular way to do "read once".